docs/protocol-methods.rst

```diff
@@ -499,6 +581,172 @@ Unsubscribe from a script hash, preventing future notifications if its :ref:`sta
   Note that :const:`False` might be returned even for something subscribed to earlier,
   because the server can drop subscriptions in rare circumstances.
 
+blockchain.outpoint.subscribe
+=============================
+
+Subscribe to a transaction outpoint (TXO), to get notifications about its status.
+A status involves up to two transactions: the funding transaction that creates
+the TXO (as one of its outputs), and the spending transaction that uses it
+as an input (spends it).
+
+**Signature**
+
+  .. function:: blockchain.outpoint.subscribe(tx_hash, txout_idx, spk_hint=None)
+  .. versionadded:: 1.5
+
+  *tx_hash*
+
+    The TXID of the funding transaction as a hexadecimal string.
+    (sometimes called prevout_hash, in inputs)
+
+  *txout_idx*
+
+    The output index, a non-negative integer. (sometimes called prevout_n, in inputs)
+
+  *spk_hint*
+
+    The scriptPubKey (output script) corresponding to the outpoint, as a hexadecimal
+    string. This is optional, and if provided might be used by the server to find
+    the outpoint. The behaviour is undefined if an incorrect value is provided.
+    Note that some implementations (such as EPS/BWT) might require this parameter
+    to be able to serve the request.
+
+.. note::  The server MAY automatically clean up subscriptions (unsubscribe the client)
+  where the spending transaction is already deeply mined at a reorg-safe height (typically
+  100+ blocks deep).
+  Similarly, the server MAY ignore new subscription requests if the spending tx is already
+  mined at a reorg-safe height but it still MUST send at least one full response.
+
+**Result**
+
+  The status of the TXO, taking the mempool into consideration.
+  The output is a dictionary, containing 0, 1, or 3 of the following items:
+
+  * *height*
+
+    The integer height of the block the funding transaction was confirmed in.
+    If the funding transaction is in the mempool; the value is
+    ``0`` if all its inputs are confirmed, and ``-1`` otherwise.
+    This key must be present if and only if there exists a funding transaction
+    (either in the best chain or in the mempool), regardless of spentness.
+
+  * *spender_txhash*
+
+    The TXID of the spending transaction as a hexadecimal string.
+    This key is present if and only if there exists a spending transaction
+    (either in the best chain or in the mempool).
+
```

Comment:
for completeness, can the input index also be returned (the index of the input on the spending transaction that spends the subscribed output)? this information is already available during lookup (when iterating through the inputs to find the previous output)

Reply:
A client can request the spender tx using `spender_txhash` in a separate request, and parse the tx to find the corresponding input index.
Do you have a use case where you would want to know the input index but not request the tx?

It is potentially quite expensive for a server to also return that here.
To give an idea, in the proposed implementation in https://github.com/spesmilo/electrumx/pull/101, these lookups are facilitated by a table in a key-value store:
```
# Key: b's' + tx_num + txout_idx
# Value: tx_num
# "which tx spent this TXO?" -- note that UTXOs are not stored.
```
`tx_num` is 5 bytes, `txout_idx` is 3 bytes. So storing the input index would turn this table from 13 to 16 bytes per item.
Alternatively, instead of storing it, the server could request the raw spender tx from bitcoind and parse it to find the input index. This looks needlessly complicated considering the client might not even need the input index.

So again, is there a good use case?

Reply:
The use case i have in mind is mempool (locally hosted mempool.space instance), it uses the input index to highlight the spending input when clicking the red arrow on an output.
The implementation i had in mind that goes through the transaction inputs in the proposed one for romanz/electrs, it stores block numbers of spending txs per txid:vout pair (8 bytes key, 4 bytes value), and when requesting the block is searched through, so the input index is available here.

This could be made an optional feature that is advertised in server.features if available